src/deploymentModals.jsx

```diff
@@ -0,0 +1,213 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+
+import cockpit from 'cockpit';
+import client from './client';
+
+import { Alert } from "@patternfly/react-core/dist/esm/components/Alert";
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Checkbox } from '@patternfly/react-core/dist/esm/components/Checkbox';
+import { Modal } from "@patternfly/react-core/dist/esm/components/Modal";
+import { Stack } from "@patternfly/react-core/dist/esm/layouts/Stack";
+import { Text } from "@patternfly/react-core/dist/esm/components/Text";
+
+import { useDialogs } from "dialogs.jsx";
+
+const _ = cockpit.gettext;
+
+export const CleanUpModal = ({ os }) => {
+    const Dialogs = useDialogs();
+
+    const [deleteTemporaryFiles, setDeleteTemporaryFiles] = useState(true);
+    const [deleteRPMmetadata, setDeleteRPMmetadata] = useState(true);
+    const [deletePendingDeployments, setDeletePendingDeployments] = useState(false);
+    const [deleteRollbackDeployments, setDeleteRollbackDeployments] = useState(false);
+    const [buttonLoading, setButtonLoading] = useState(false);
+    const [error, setError] = useState("");
+
+    const doCleanup = () => {
+        setButtonLoading(true);
+        setError("");
+
+        const cleanupFlags = [];
+        if (deleteTemporaryFiles) {
+            cleanupFlags.push("base");
+        }
+        if (deleteRPMmetadata) {
+            cleanupFlags.push("repomd");
+        }
+        if (deletePendingDeployments) {
+            cleanupFlags.push("pending-deploy");
+        }
+        if (deleteRollbackDeployments) {
+            cleanupFlags.push("rollback-deploy");
+        }
+
+        return client.run_transaction("Cleanup", [cleanupFlags], os)
+                .then(Dialogs.close)
+                .catch(ex => {
+                    console.warn(ex);
+                    setError(ex.message);
+                    setButtonLoading(false);
```

Comment:
These 4 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231128-144720-91f0c34a-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)